From: "Gustavo A. R. Silva" <gustavo@embeddedor.com>
Date: Thu, 28 Jun 2018 08:08:09 -0500
Subject: wlcore: Fix memory leak in wlcore_cmd_wait_for_event_or_timeout
Git-commit: 04614fe46f315a223d96b8f5df7b52130965677b
Patch-mainline: v4.19-rc1
References: git-fixes

In case memory resources for *events_vector* were allocated, release
them before return.

Addresses-Coverity-ID: 1470194 ("Resource leak")
Fixes: 4ec7cece87b3 ("wlcore: Add missing PM call for wlcore_cmd_wait_for_event_or_timeout()")
Signed-off-by: Gustavo A. R. Silva <gustavo@embeddedor.com>
Acked-by: Tony Lindgren <tony@atomide.com>
Signed-off-by: Kalle Valo <kvalo@codeaurora.org>
Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 drivers/net/wireless/ti/wlcore/cmd.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/net/wireless/ti/wlcore/cmd.c
+++ b/drivers/net/wireless/ti/wlcore/cmd.c
@@ -194,7 +194,7 @@ int wlcore_cmd_wait_for_event_or_timeout
 
 	ret = wl1271_ps_elp_wakeup(wl);
 	if (ret < 0)
-		return ret;
+		goto free_vector;
 
 	do {
 		if (time_after(jiffies, timeout_time)) {
@@ -228,6 +228,7 @@ int wlcore_cmd_wait_for_event_or_timeout
 
 out:
 	wl1271_ps_elp_sleep(wl);
+free_vector:
 	kfree(events_vector);
 	return ret;
 }
